subprojects/kotlin-compiler-embeddable/kotlin-compiler-embeddable.gradle.kts

```diff
@@ -55,9 +55,8 @@ tasks {
         dependenciesIncludes.set(mapOf(
             "jansi-" to listOf("META-INF/native/**", "org/fusesource/jansi/internal/CLibrary*.class")
         ))
-        additionalFiles = fileTree(classpathManifest.get().manifestFile.parentFile) {
-            include(classpathManifest.get().manifestFile.name)
-        }
+        additionalFiles = files(classpathManifest).asFileTree
```

Comment:
What happens if we don't use `asFileTree`?

Reply:
`files()` returns `ConfigurableFileCollection`:

`ConfigurableFileCollection extends FileCollection extends Iterable<File>, AntBuilderAware, Buildable`

Reply:
I guess I asked the wrong question: Why is `additionalFiles` of type `FileTree`. I think it is so that `@Classpath` input tracking does not consider the classpath manifest as a classpath entry, but the parent directory as a classpath entry.

I don't even know why we use `@Classpath` on additionalFiles, given that the inputs should probably not be ignored. `@PathSensitive(PathSensitivity(NAME_ONLY)` would suffice I guess.

Anyway, let's leave it as is for now. We should inform the Kotlin-DSL team for sure.

Reply:
OK. @gradle/kotlin-dsl would you like to take care of the followups?

Reply:
Good catch @blindpirate!
You're right @wolfs, this is less than ideal.
Here's a PR that should make things better, please take a look https://github.com/gradle/gradle/pull/8745